flaky unit test in pkg/util/wsstream/

@smarterclayton 

```
2015/10/23 17:11:34 http: panic serving 127.0.0.1:40414: bad read
goroutine 69 [running]:
net/http.(*conn).serve.func1(0xc82001c840, 0x2b0127787228, 0xc82003a1a0)
	/usr/lib/google-golang/src/net/http/server.go:1287 +0xb5
k8s.io/kubernetes/pkg/util/wsstream.(*errorReader).Read(0xc820780080, 0xc8207d6800, 0x800, 0x800, 0xc8207744e0, 0x0, 0x0)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wsstream/stream_test.go:151 +0xcc
k8s.io/kubernetes/pkg/util/wsstream.messageCopy(0xc820786240, 0x2b012c44a000, 0xc820780080, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wsstream/stream.go:105 +0x1dd
k8s.io/kubernetes/pkg/util/wsstream.(*Reader).handle(0xc820776150, 0xc820786240)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wsstream/stream.go:86 +0x196
k8s.io/kubernetes/pkg/util/wsstream.readWebSocket.func1(0xc820786240)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wsstream/stream_test.go:171 +0xcf
golang.org/x/net/websocket.Server.serveWebSocket(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x95c390, ...)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/Godeps/_workspace/src/golang.org/x/net/websocket/server.go:89 +0x2d2
golang.org/x/net/websocket.Handler.ServeHTTP(0xc820776180, 0x2b01277874e0, 0xc8207d4000, 0xc8207cc000)
	/usr/local/google/home/thockin/src/go/src/k8s.io/kubernetes/Godeps/_workspace/src/golang.org/x/net/websocket/server.go:112 +0x83
net/http/httptest.(*waitGroupHandler).ServeHTTP(0xc820774220, 0x2b01277874e0, 0xc8207d4000, 0xc8207cc000)
	/usr/lib/google-golang/src/net/http/httptest/server.go:200 +0xc2
net/http.serverHandler.ServeHTTP(0xc82079e120, 0x2b01277874e0, 0xc8207d4000, 0xc8207cc000)
	/usr/lib/google-golang/src/net/http/server.go:1862 +0x19e
net/http.(*conn).serve(0xc82001c840)
	/usr/lib/google-golang/src/net/http/server.go:1361 +0xbee
created by net/http.(*Server).Serve
	/usr/lib/google-golang/src/net/http/server.go:1910 +0x3f6
--- FAIL: TestStreamClosedDuringRead (0.02s)
	stream_test.go:131: unexpected server read: []
FAIL
FAIL	k8s.io/kubernetes/pkg/util/wsstream	0.207s

--- a/pkg/util/wsstream/stream_test.go
+++ b/pkg/util/wsstream/stream_test.go
@@ -110,28 +110,31 @@ func TestStreamSurvivesPanic(t *testing.T) {
 }
 
 func TestStreamClosedDuringRead(t *testing.T) {
-	ch := make(chan struct{})
-	input := "some random text"
-	errs := &errorReader{
-		reads: [][]byte{
-			[]byte("some random"),
-			[]byte(" text"),
-		},
-		err:   fmt.Errorf("stuff"),
-		pause: ch,
-	}
-	r := NewReader(errs, false)
+	for i := 0; i < 25; i++ {
+		ch := make(chan struct{})
+		input := "some random text"
+		errs := &errorReader{
+			reads: [][]byte{
+				[]byte("some random"),
+				[]byte(" text"),
+			},
+			err:   fmt.Errorf("stuff"),
+			pause: ch,
+		}
+		r := NewReader(errs, false)
 
-	data, err := readWebSocket(r, t, func(c *websocket.Conn) {
-		c.Close()
-		time.Sleep(time.Millisecond)
-		close(ch)
-	})
-	if !reflect.DeepEqual(data, []byte(input)) {
-		t.Errorf("unexpected server read: %v", data)
-	}
-	if err == nil || !strings.Contains(err.Error(), "use of closed network connection") {
-		t.Fatal(err)
+		data, err := readWebSocket(r, t, func(c *websocket.Conn) {
+			c.Close()
+			close(ch)
+		})
+		// verify that the data returned by the server on an early close always has a specific error
+		if err == nil || !strings.Contains(err.Error(), "use of closed network connection") {
+			t.Fatal(err)
+		}
+		// verify that the data returned is a strict subset of the input
+		if !bytes.HasPrefix([]byte(input), data) && len(data) != 0 {
+			t.Fatalf("unexpected server read: %q", string(data))
+		}
 	}
 }
 

